src/sonic-frr/patch/0037-bgpd-Ensure-community-data-is-freed-in-some-cases.patch

```diff
@@ -0,0 +1,246 @@
+Code inspection of 8.5.1 -vs- latest master shows the same problem
+exists.  Code should be able to be followed from there to here.
+
+What is happening:
+
+There is a route-map being applied that modifes the outgoing community
+to a peer.  This is saved in the attr copy created in
+subgroup_process_announce_selected.  This community pointer is not
+interned.  So the community->refcount is still 0.  Normally when
+a prefix is announced, the attr and the prefix are placed on a
+adjency out structure where the attribute is interned.  This will
+cause the community to be saved in the community hash list as well.
+In a non-normal operation when the decision to send is aborted after
+the route-map application, the attribute is just dropped and the
+pointer to the community is just dropped too, leading to situations
+where the memory is leaked.  The usage of bgp suppress-fib would
+would be a case where the community is caused to be leaked.
+Additionally the previous commit where an unsuppress-map is used
+to modify the outgoing attribute but since unsuppress-map was
+not considered part of outgoing policy the attribute would be dropped as
+well.  This pointer drop also extends to any dynamically allocated
+memory saved by the attribute pointer that was not interned yet as well.
+
+So let's modify the return case where the decision is made to
+not send the prefix to the peer to always just flush the attribute
+to ensure memory is not leaked.
+
+Fixes: #15459
+Signed-off-by: Donald Sharp <sharpd@nvidia.com>
+
+update
+---
+ bgpd/bgp_conditional_adv.c |  5 +++--
+ bgpd/bgp_route.c           | 20 ++++++++++++++------
+ bgpd/bgp_updgrp.h          |  2 +-
+ bgpd/bgp_updgrp_adv.c      | 35 ++++++++++++++++++++++-------------
+ 4 files changed, 40 insertions(+), 22 deletions(-)
+
+diff --git a/bgpd/bgp_conditional_adv.c b/bgpd/bgp_conditional_adv.c
+index 4ad00ed121..89ea85ff46 100644
+--- a/bgpd/bgp_conditional_adv.c
++++ b/bgpd/bgp_conditional_adv.c
+@@ -134,8 +134,9 @@ static void bgp_conditional_adv_routes(struct peer *peer, afi_t afi,
+ 			if (update_type == UPDATE_TYPE_ADVERTISE &&
+ 			    subgroup_announce_check(dest, pi, subgrp, dest_p,
+ 						    &attr, &advmap_attr)) {
+-				bgp_adj_out_set_subgroup(dest, subgrp, &attr,
+-							 pi);
++				if (!bgp_adj_out_set_subgroup(dest, subgrp,
++							      &attr, pi))
++					bgp_attr_flush(&attr);
+ 			} else {
+ 				/* If default originate is enabled for
+ 				 * the peer, do not send explicit
+diff --git a/bgpd/bgp_route.c b/bgpd/bgp_route.c
+index 2a1ca2b2e8..7414524fe3 100644
+--- a/bgpd/bgp_route.c
++++ b/bgpd/bgp_route.c
+@@ -2917,16 +2917,24 @@ void subgroup_process_announce_selected(struct update_subgroup *subgrp,
+ 			 * in FIB, then it is advertised
+ 			 */
+ 			if (advertise) {
+-				if (!bgp_check_withdrawal(bgp, dest))
+-					bgp_adj_out_set_subgroup(
+-						dest, subgrp, &attr, selected);
+-				else
++				if (!bgp_check_withdrawal(bgp, dest)) {
```

Comment:
Can you please check the latest update on https://github.com/FRRouting/frr/pull/15466 ? The diff has changed in this file

Reply:
> Can you please check the latest update on [FRRouting/frr#15466](https://github.com/FRRouting/frr/pull/15466) ? The diff has changed in this file

HI, @dgsudharsan thanks for reviewing the code. This part of the code should be the same, the difference should be related to the difference between 202305 branch and master. thanks.